core/src/main/java/org/elasticsearch/Build.java

```diff
@@ -79,10 +79,10 @@ static URL getElasticsearchCodebase() {
         return Build.class.getProtectionDomain().getCodeSource().getLocation();
     }
 
-    private String shortHash;
-    private String date;
+    private final String shortHash;
+    private final String date;
 
-    Build(String shortHash, String date, boolean isSnapshot) {
+    public Build(String shortHash, String date, boolean isSnapshot) {
```

Comment:
I was trying to avoid this but need to have a way to create this object on the parsing side. Any ideas how to do this differently are welcome.